main.js

```diff
@@ -389,10 +443,10 @@ function initializeDeviceObjects(deviceId, data, callback) {
                     id: '.info.address',
                     type: 'string',
                 });
-
+                let padlen = data.DataRecord.length.toString().length;
                 for (let i = 0; i < data.DataRecord.length; i++) {
                     currentState = {};
-                    currentState.id = '.data.' + data.DataRecord[i].id;
+                    currentState.id = '.data.' + data.DataRecord[i].id.toString().padStart(padlen, '0');
```

Comment:
This is a brealing change because Object IDs are changed, correct? hmmmmm .... is it really that beneficial to risk a breaking change?

Reply:
And "padStart" is not compatible with nodejs 6 so it is the second breaking chnagfe because it will kill nodejs 6 support. I know it is convenient, but can we work around that one?

Reply:
This part... I probably should not have included this in the same PR.

I think generally speaking that changing the object IDs is a good thing because even though M-Bus should be a self describing format you usually need to look at the telegram description in the meter's manual to really know what values represent which data. (At least if the meter is a bit more complicated and therefore outputs more than a few values.) So the order of the values becomes important when checking against the examples data from manuals.

[I am bit doubtful about another aspect of this part: **If** for some reason the length of the telegram changes, the padding will be wrong and the IDs will not match at all, even it there was some data that matches. However, the adapter relies on unchanged telegram at many points anyway, so I think that is not really a problem...]

Breaking existing installations is of course also a problem. However, this whole PR *might* break an installation anyway, if the units associated with the states are not the same as they are assigned with my patch. I think the most likely candidates are energy values in kWh and there is a good chance that turning on "force kWh" will get you a compatible result.

A replacement for padStart should not be a problem https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart#Polyfill

Reply:
The padding, to be hones, is only a convenice thing because normally object IDs are only unique IDs and the only thing where it helps is that the list in Admin UI is correctly ordered and not 1,10,11 cases comeup ... but in this case the question is if the padding is really added that much benefit. Also when checking with manual you only need to read the numbers correctly :-)

When the meter changes the reported values then the probim will be there in any case - but this is caused by the protocol because the only order is the order how data are send out. So I think this will never change for a meter.

For Units: Can we not have backward compatible settings? Or, should not "forcekWh=false" cause a compatible setting? When unit values are set more correct then I'm fine with it or other parsing corrected other stuff.